pilot/pkg/config/kube/crdclient/cache_handler.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crdclient
```

Comment:
are there any changes here or is this just a simple code move?

Reply:
Its largely the same but the struct has a slightly different structure (as we have different info now) and onEvent is slightly different from the change in how we translate objects. But largely the same